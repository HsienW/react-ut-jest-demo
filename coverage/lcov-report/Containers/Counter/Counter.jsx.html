<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Containers/Counter/Counter.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">Containers/Counter</a> Counter.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/5</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {CounterRedux} from  '../../Redux/Modules';
&nbsp;
class Counter extends Component {
&nbsp;
    state = {
        count: 0
    };
&nbsp;
    addCount = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >        this.props.CounterActionsCreator.addNumber();</span>
    };
&nbsp;
<span class="fstat-no" title="function not covered" >    re</span>nder() {
<span class="cstat-no" title="statement not covered" >        return (</span>
            &lt;div&gt;
                &lt;span&gt;{`當前數字: ${this.state.count}`}&lt;/span&gt;
                &lt;br /&gt;
                &lt;button type="button" onClick={this.addCount}&gt;+ 1&lt;/button&gt;
            &lt;/div&gt;
        );
    }
}
&nbsp;
Counter.propTypes = {
    CounterActionsCreator: PropTypes.object.isRequired,
};
&nbsp;
export default connect(
<span class="fstat-no" title="function not covered" >    (s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >        return {action: state.CounterRedux.action};</span>
    },
<span class="fstat-no" title="function not covered" >    (d</span>ispatch) =&gt; {
<span class="cstat-no" title="statement not covered" >        return {</span>
            CounterActionsCreator: bindActionCreators(CounterRedux.CounterActions, dispatch),
        };
    }
)(Counter);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Nov 30 2019 20:56:09 GMT+0800 (GMT+08:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
