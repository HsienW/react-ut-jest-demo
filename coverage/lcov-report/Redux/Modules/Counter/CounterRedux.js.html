<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Redux/Modules/Counter/CounterRedux.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">Redux/Modules/Counter</a> CounterRedux.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {createAction} from 'redux-actions';
&nbsp;
export const CounterActions = {
    addCounter: 'ADD_COUNTER',
};
&nbsp;
const addNumber = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    return <span class="fstat-no" title="function not covered" >(d</span>ispatch) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        dispatch(createAction(CounterActions.addCounter)(</span>
            {
                type: CounterActions.addCounter,
                payload: {number: 1,},
            }
        ));
    };
};
&nbsp;
export const CounterActionsCreator = {
    addNumber,
};
&nbsp;
export default function <span class="fstat-no" title="function not covered" >CounterReducer(</span>state = <span class="branch-0 cbranch-no" title="branch not covered" >{action: ''},</span> action) {
<span class="cstat-no" title="statement not covered" >    switch (action.type) {</span>
        case CounterActions.addCounter:
<span class="cstat-no" title="statement not covered" >            return {...state, count: state.count + 1};</span>
&nbsp;
        default:
<span class="cstat-no" title="statement not covered" >            return state;</span>
    }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Nov 30 2019 20:56:09 GMT+0800 (GMT+08:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
