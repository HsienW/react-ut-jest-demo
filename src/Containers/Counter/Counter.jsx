import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {CounterRedux} from  '../../Redux/Modules';

export class Counter extends Component {

    state = {
        count: 0
    };

    addCount = () => {
        this.props.CounterActionsCreator.addNumber();
    };

    render() {
        return (
            <div>
                <span>{`當前數字: ${this.state.count}`}</span>
                <br />
                <button type="button" onClick={this.addCount}>+ 1</button>
            </div>
        );
    }
}

Counter.propTypes = {
    CounterActionsCreator: PropTypes.object.isRequired,
};

export default connect(
    (state) => {
        return {action: state.CounterRedux.action};
    },
    (dispatch) => {
        return {
            CounterActionsCreator: bindActionCreators(CounterRedux.CounterActions, dispatch),
        };
    }
)(Counter);
