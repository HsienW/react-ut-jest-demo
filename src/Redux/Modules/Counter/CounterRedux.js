import {createAction} from 'redux-actions';

export const CounterActions = {
    addCounter: 'ADD_COUNTER',
};

const addNumber = () => {
    return (dispatch) => {
        dispatch(createAction(CounterActions.addCounter)(
            {
                type: CounterActions.addCounter,
                payload: {number: 1,},
            }
        ));
    };
};

export const CounterActionsCreator = {
    addNumber,
};

export default function CounterReducer(state = {action: ''}, action) {
    switch (action.type) {
        case CounterActions.addCounter:
            return {...state, count: state.count + 1};

        default:
            return state;
    }
}
