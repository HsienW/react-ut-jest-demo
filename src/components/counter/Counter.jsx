import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {CounterActions} from '../../Redux/Modules/CounterRedux';

export class Counter extends Component {

    state = {
        count: 0
    };

    increment = () => {
        this.setState({
            count: this.state.count + 1
        });
    };

    decrement = () => {
        this.setState({
            count: this.state.count - 1
        });
    };

    render() {
        return (
            <div>
                <div className='number'>{this.state.count}</div>
                <br/>
                <button className='increment' onClick={this.increment}>+1</button>
                <button className='decrement' onClick={this.decrement}>-1</button>
            </div>
        );
    }
}


Counter.propTypes = {
    counterTwo: PropTypes.number,
};


export default connect(
    (state) => {
        return {action: state.CounterReducer.action};
    },
    (dispatch) => {
        return {
            CounterActionsCreator: bindActionCreators(CounterActions, dispatch),
        };
    }
)(Counter);
