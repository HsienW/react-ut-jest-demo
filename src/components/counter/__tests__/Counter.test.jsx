import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Counter} from '../Counter';


Enzyme.configure({adapter: new Adapter()});

describe('Counter component', () => {
    it('should created', () => {
        const component = Enzyme.shallow(<Counter/>);
        expect(component).toMatchSnapshot();
    });

    it('should render component', () => {
        const component = Enzyme.mount(<Counter/>);
        expect(component.find('p').length).toEqual(1);
        expect(component.find('.increment').length).toEqual(1);
        expect(component.find('.decrement').length).toEqual(1);
    });
});

describe('increment function testing', () => {
    const component = Enzyme.mount(<Counter/>);
    const counterWrapper = component.find(Counter);

    it('count value should equal 1', () => {
        counterWrapper.find('button.increment').simulate('click');
        expect(counterWrapper.find('p').text()).toEqual('1');
    });
});

describe('decrement function testing', () => {
    const component = Enzyme.mount(<Counter/>);
    const counterWrapper = component.find(Counter);

    it('count value should equal -1', () => {
        component.find('button.decrement').simulate('click');
        expect(counterWrapper.find('p').text()).toEqual('-1');
    });
});