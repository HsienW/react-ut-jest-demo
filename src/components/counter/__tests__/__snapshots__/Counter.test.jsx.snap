// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should created 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          0
        </p>,
        <br />,
        <button
          className="increment"
          onClick={[Function]}
        >
          +1
        </button>,
        <button
          className="decrement"
          onClick={[Function]}
        >
          -1
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+1",
          "className": "increment",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+1",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-1",
          "className": "decrement",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "-1",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            0
          </p>,
          <br />,
          <button
            className="increment"
            onClick={[Function]}
          >
            +1
          </button>,
          <button
            className="decrement"
            onClick={[Function]}
          >
            -1
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+1",
            "className": "increment",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+1",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-1",
            "className": "decrement",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "-1",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
